Bio/Align/__init__.py

```diff
@@ -971,7 +971,17 @@ def alignment(self):
             lines = [str(record.seq) for record in records]
             coordinates = Alignment.infer_coordinates(lines)
             for record in records:
-                record.seq = record.seq.replace("-", "")
+                if record.letter_annotations:
+                    indices = [i for i, c in enumerate(record.seq) if c != "-"]
+                    letter_annotations = dict(record.letter_annotations)
+                    record.letter_annotations.clear()
+                    record.seq = record.seq.replace("-", "")
+                    for key, value in letter_annotations.items():
+                        value = "".join([value[i] for i in indices])
```

Comment:
This will convert any per-letter-annotation value into a string, but it might have been a list or tuple (e.g. quality scores).